### PR TITLE
feat: wishlist context added with api call

### PR DESCRIPTION
**Pull request by humzahasan**
PR Description : 

PR contains changes made for wishlist context and changes in UI on Wishlist, Product listing and Cart page. 

Please Review !

### CODE REVIEW

File: src/context/wishlist-context.js [hunk 1]
```diff
@@ -1,14 +1,89 @@
...
+        headers: {
+          authorization: encodedToken,
+        },
+      });
+
+      if (res.status === 201) {
+        setWishlist(res.data);
+      }
+    } catch (error) {
+      toast.error("Oh no, failed to get your cart!");
```
Comment (rutu14) [inline src/context/wishlist-context.js:38]: Since it is in wishlist function the error msg can be changed.

**Review by rutu14 — APPROVED** [review]
The code is good to merge.
UI looks great.

**Review by archit7448 — APPROVED** [review]
- NICE UI 
- code structure is proper 
- console.log can be remove 
- looks good to merge

**Review by Ranjeetk489 — APPROVED** [review]
UI looks great good to merge

**Pull request merged**